### PR TITLE
[main][feat] split heath check logic off from DeltaNeutralVault

### PR DESCRIPTION
**Pull request by alpanaca**
## Description
Move the health check logic out of DeltaNeutralVault contract due to size limit

## Why?
With the current size, it's almost impossible to add more feature to the Delta Neutral Vault

## ChangeLogs:
- HealthChecker contract will be responsible for deposit , withdraw health check
- Existing Delta Neutral Vault will need to set the address of the HealthChecker in order to be operable
- setter function in Delta Neutral Vault

### CODE REVIEW

**Review by abc-alpaca — DISMISSED** [review]

File: solidity/contracts/8.13/DeltaNeutralVault04HealthChecker.sol [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+  ) external view {
+    uint256 _toleranceBps = _config.positionValueTolerance();
+    uint8 _leverageLevel = _config.leverageLevel();
+
+    // 1. check if vault accept new total position value
+    if (!_isVaultSizeAcceptable(_positionInfoAfter, _config)) {
+      revert DeltaNeutralVault04HealthChecker_PositionValueExceedLimit();
+    }
+    // 2. check equity allocation
+
```
Comment (alpanaca) [inline solidity/contracts/8.13/DeltaNeutralVault04HealthChecker.sol:77]:
```suggestion
```

File: solidity/contracts/8.13/interfaces/IDeltaNeutralStruct.sol [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    uint256 assetPositionEquity;
+    uint256 assetPositionDebtValue;
+    uint256 assetLpAmount;
+  }
+
+  struct Outstanding {
+    uint256 stableAmount;
+    uint256 assetAmount;
+    uint256 nativeAmount;
+  }
```
Comment (alpanaca) [inline solidity/contracts/8.13/interfaces/IDeltaNeutralStruct.sol:30]:
```suggestion
```

File: solidity/contracts/8.13/DeltaNeutralVault04.sol [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+  /// @param _data The calldata to pass along to the proxy action for more working context.
+  function deposit(
+    uint256 _stableTokenAmount,
+    uint256 _assetTokenAmount,
+    address _shareReceiver,
+    uint256 _minShareReceive,
+    bytes calldata _data
+  ) public payable onlyEOAorWhitelisted collectFee nonReentrant returns (uint256) {
+    PositionInfo memory _positionInfoBefore = positionInfo();
+    Outstanding memory _outstandingBefore = _outstanding();
```
Comment (alpanaca) [inline solidity/contracts/8.13/DeltaNeutralVault04.sol:284]:
```suggestion
```

File: solidity/contracts/8.13/DeltaNeutralVault04.sol [hunk 4]
```diff
@@ -0,0 +1,727 @@
...
+  function withdraw(
+    uint256 _shareAmount,
+    uint256 _minStableTokenAmount,
+    uint256 _minAssetTokenAmount,
+    bytes calldata _data
+  ) external onlyEOAorWhitelisted collectFee nonReentrant returns (uint256) {
+    if (_shareAmount == 0) revert DeltaNeutralVault04_InvalidShareAmount();
+
+    PositionInfo memory _positionInfoBefore = positionInfo();
+    Outstanding memory _outstandingBefore = _outstanding();
```
Comment (alpanaca) [inline solidity/contracts/8.13/DeltaNeutralVault04.sol:348]:
```suggestion
```

**Review by alpanaca — COMMENTED** [review]
Please remove all outstanding related

**Review by jr-alpaca — APPROVED** [review]

**Review by abc-alpaca — APPROVED** [review]

**Pull request merged**